pkg/skaffold/util/util.go

```diff
@@ -154,6 +154,66 @@ func ReadConfiguration(filename string) ([]byte, error) {
 	}
 }
 
+// A type holding configuration properties
+type ConfigProperties map[string]string
+
+// Tests if a file exists
+func FileExists(filepath string) bool {
+	if _, err := os.Stat(filepath); !os.IsNotExist(err) {
+		if err == nil {
+			return true
+		}
+	}
+	return false
+}
+
+var defaultPropsPaths = []string{"/etc/skaffold/", "$HOME/.skaffold"}
+
+// Reads an environment file that contains a set of environment variables
+// The file is expected to be in YAML. Once read, the function will
+// return an object that contains all the properties defined in the file
+func ReadEnvironmentConfigProperties(pathToEnvFile string) (*viper.Viper, error) {
+
+	logrus.Debugf("Will load environment file from :%s", pathToEnvFile)
+	configuration := viper.New()
+	for _, path := range defaultPropsPaths {
+		configuration.AddConfigPath(path)
```

Comment:
could you explain how this works? it looks like you're adding default paths that the config files *might* live in, but then below you're explicitly calling `configuration.ReadConfig()` on a file descriptor created from the provided `pathToEnvFile`. are these other config files that might also get read if they exist?